lib/logstash/codecs/fluent.rb

```diff
@@ -145,4 +148,10 @@ def generate_event(map, fluent_time, tag)
     event
   end
 
+  def map_timestamp(event_hash)
+    event_hash.each do |k, v|
```

Comment:
Unfortunately this only works for top level timestamp fields, something like

```
filter {
  ruby { code => "event.set('[logstash][timestamp]', Time.now());" }
}
```

would still fail in the same way

Reply:
Ah, thanks for catching out. I thought `LogStash::Util.normalize(event.to_hash)` L77 will flat the structure. Fixed with [this commit](https://github.com/logstash-plugins/logstash-codec-fluent/pull/30/commits/aba794f699094872e5dbc26ca2ad8c2aa2aca16c).